hw/ip/otbn/doc/_index.md

```diff
@@ -490,6 +490,35 @@ The error that caused the alert can be determined by reading the {{< regref "FAT
 If OTBN was running, this value will also be reflected in the {{< regref "ERR_BITS" >}} register.
 A fatal alert can only be cleared by resetting OTBN through the `rst_ni` line.
 
+
+### Error Prioritization
+
+OTBN can signal multiple errors at once, by setting multiple bits of {{< regref "ERR_BITS" >}}.
+There are some scenarios where one error takes priority over another.
+In these scenarios the higher priority error will suppress some other errors.
+
+* Any error that produces a fatal alert will suppress all non fatal errors.
+* Any `call_stack` error that occurs due to a pop will suppress:
+  - `bad_data_addr`
+  - `bad_insn_addr`
+  - `loop`
+* Any `illegal_insn` error will suppress:
+  - `bad_data_addr`
+  - `bad_insn_addr`
+  - `call_stack`
+  - `loop`
+
+(See {{< regref "ERR_BITS" >}} for the error meanings).
+
+Some combinations of error bits **cannot occur** in a correctly functioning OTBN implementation.
+Should these combinations be observed the `fatal_bad_err` bit in {{< regref "ERR_BITS" >}} will be set, producing a fatal alert.
```

Comment:
Going forward, we will have more cases like this where we detect problems with "internal consistency" of OTBN (e.g. if we detect an invalid state transition). Do we want to have a separate fatal error for all those cases, or should we just combine them into a single "fatal_corrupt_state" (or something along those lines) to avoid adding more and more error codes?

Another way to think about it: 
- Would software on Ibex want to handle these cases separately, or would the mitigation always be the same?
- Do we possibly even give away information to an attacker if the error codes are too detailed?

I'm trending to a single error code for all those consistency errors.

Reply:
Yes I think we should have a single error code and start doing some more internal consistency checks (in a sense assertions we synthesize and include in the netlist).

Having written this and thought about it I think we should drop all notion of *Cannot Occur* errors from this documentation. It's not programmer visible and confusing as we've got another sense of cannot occur in things that architectural shouldn't happen but do in RTL and then we suppress from the programmer reported error bits.

Instead we'll just have a general `fatal_corrupt_state` or maybe `fatal_consistency_error` (as it's not just about state) which will include these 'assertions' around the internal error signals.

Some form of what I've written here will become RTL comments in the appropriate place.